projects/power_distribution/src/pd_fan_ctrl.c

```diff
@@ -56,7 +57,12 @@ static void prv_front_temp_to_fan_percent(uint16_t v_measured, uint8_t *fan_spee
     *fan_speed = 100;
     return;
   }
-  double ratio = s_fan_storage.ref_reading / 100.0;
+  if ((v_measured == 0) || (v_measured > (FRONT_FAN_CTRL_MAX_VALUE_MV + ADC_EPSILON_MV))) {
+    *fan_speed = 0;
+    return;
+  }
+
+  double ratio = FRONT_FAN_CTRL_MAX_VALUE_MV / 100.0;
   *fan_speed = v_measured / ratio;
```

Comment:
Actually wait I think that if `v_measured > FRONT_FAN_CTRL_MAX_VALUE_MV`, but `v_measured < FRONT_FAN_CTRL_MAX_VALUE_MV + ADC_EPSILON_MV`, we could end up outputting a fan speed over 100. Could you `MIN()` the calculated fan speed with 100 to make sure that doesn't happen?

Reply:
Sure, I'm guessing I don't have to worry about negative voltages, or swapped polarities